e-commerce/src/components/forms/userProfile/userProfile.tsx

```diff
@@ -0,0 +1,265 @@
+import React, { useState, useEffect } from 'react';
+import Modal from 'react-modal';
+import { useNavigate } from 'react-router-dom';
+import toast, { Toaster } from 'react-hot-toast';
+import { useAppDispatch } from '../../../redux/hooks';
+import {
+  setUserLogged,
+  logout,
+  userInitial,
+} from '../../../redux/store/userSlice';
+import store from '../../../redux/store/store';
+import TextField from '../elements/textField';
+import ChangePassword from '../floatForms/changePassword';
+import UserMainData from '../floatForms/userMainData';
+import { EditAddress } from '../floatForms/editAddress';
+import { AddressesList } from './addressesList';
+/* API */
+import { getCustomerInfo } from '../../../services/api/getCustomerInfo';
+import { deleteAddress } from '../../../services/api/changeAddresses';
+
+function UserProfile(): React.ReactElement {
+  const [userData, setUserData] = useState(userInitial);
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+
+  async function updatesetUserData() {
+    const user = await getCustomerInfo(true);
+    if (user) setUserData(user);
+  }
+
+  const showToast = ({
+    message,
+    thisError,
+  }: {
+    message: string;
+    thisError: boolean;
+  }) => {
+    updatesetUserData();
+    if (!thisError) {
+      toast.success(message, {
+        style: {
+          border: '1px solid #713200',
+          padding: '16px',
+          color: 'white',
+          backgroundColor: 'green',
+        },
+        iconTheme: {
+          primary: 'white',
+          secondary: 'green',
+        },
+      });
+    } else {
+      toast.error(message, {
+        style: {
+          border: '1px solid #713200',
+          padding: '16px',
+          color: 'red',
+          backgroundColor: 'pink',
+        },
+        iconTheme: {
+          primary: 'white',
+          secondary: 'red',
+        },
+      });
+    }
+  };
+
+  const [modalPasswordIsOpen, setModalPasswordIsOpen] = useState(false);
```

Comment:
Посмотрите такой хук, useToggle
```suggestion
  const [isModalPasswordOpen, setIsModalPasswordOpen] = useState(false);
```